source/material_model/rheology/elasticity.cc

```diff
@@ -252,11 +252,33 @@ namespace aspect
             for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
               {
                 // Get old stresses from compositional fields
+                // TODO: use $tau^{t+\Delta te}$ or evaluate the old solution for stress
+                // (and viscosity)
+                // Explanation: for the assembly of the Stokes equations (the force term goes
+                // into the rhs of the momentum equation), this function is evaluated.
+                // This is after the advection equations have been solved, and hence in.composition
+                // contains the rotated, averaged and advected stresses at $tau^{t+\Delta t}$.
+                //
+                // For an iterative Advection scheme, in.composition could even be different
+                // every time the Stokes system is assembled.
+                //
+                // Moresi et al. (2003) use the full stresses at $t$ (Eq.  30) in the force term.
+                // There is also an incorrect minus sign in the force term.
+                // The force term should be computed as:
+                // $\frac{-\eta_{effcreep}  tau_0}{\eta_{e}}$, where $\eta_{effcreep}$ is the
+                // harmonic average of the vp and e viscosity.
+                // How will we get to this viscosity? 
                 SymmetricTensor<2,dim> stress_old;
                 for (unsigned int j=0; j < SymmetricTensor<2,dim>::n_independent_components; ++j)
                   stress_old[SymmetricTensor<2,dim>::unrolled_to_component_indices(j)] = in.composition[i][j];
 
                 // Average viscoelastic viscosity
+                // TODO: use the viscosity corresponding to the stresses selected above.
+                // Explanation: out.viscosities is computed during the assembly of the Stokes equations
+                // based on the current_linearization_point. This means that it will be updated after every
+                // nonlinear Stokes iteration, and is ahead of the stresses that are used in the force term.
+                // Also, out.viscosities contains the vep viscosity, which is either the harmonic average
+                // of the viscous and elastic viscosity, or the plastic viscosity, not $\eta_{effcreep}$.
```

Comment:
As noted above, until we start using a local iterative method, I think the viscosity value from the previous time step should be correct?

Reply:
@naliboff If we use the old viscosities from the previous time step, then we could have elements that are spuriously yielding just because they yielded in the previous time step. 

> the harmonic average of the viscous and elastic viscosity, or the plastic viscosity, not $\eta_{effcreep}$.
@anne-glerum if the plastic damper is not being used, these are the same, aren't they? If the material is yielding,  then eta_effcreep is the plastic viscosity. If it isn't yielding, then eta_effcreep is the harmonic average of the viscous and elastic viscosity.

Reply:
Yesterday, eta_effcreep in the Appendix seemed to be something else, but I see you've updated it now. So yes, out.viscosities returns either the harmonic average of the viscous and elastic viscosity, or the plastic viscosity, which is what is needed for eta_effcreep.

Reply:
Yes, sorry about that... I copy-pasted from the previous section and didn't change everything I should have changed. I think the Appendices should be error free now <crosses fingers>.